### PR TITLE
Updated ReflectionPropertyAccessor

### PR DESCRIPTION
**Pull request by jeremy-smith-maco**
Now checks parents for the given property.

### CODE REVIEW

File: src/PropertyAccess/ReflectionPropertyAccessor.php [hunk 1]
```diff
@@ -114,9 +109,33 @@ private function propertyExists($objectOrArray, $propertyPath)
...
+
+    /**
+     * Finds which class defines the property.
+     *
+     * @param mixed  $class
+     * @param string $property
+     *
+     * @return string
+     */
+    private function findClass($class, $property)
```
Comment (theofidry) [inline src/PropertyAccess/ReflectionPropertyAccessor.php]: it also returns `null` if no class is found

File: src/PropertyAccess/ReflectionPropertyAccessor.php [hunk 2]
```diff
@@ -114,9 +109,33 @@ private function propertyExists($objectOrArray, $propertyPath)
 
         $reflectionClass = (new ReflectionClass(get_class($objectOrArray)));
 
-        return (
-            $reflectionClass->hasProperty($propertyPath)
-            && false === $reflectionClass->getProperty($propertyPath)->isStatic()
-        );
+        while ($reflectionClass)
+        {
```
Comment (theofidry) [inline src/PropertyAccess/ReflectionPropertyAccessor.php]: CS, let's keep the braket on the same line

File: src/PropertyAccess/ReflectionPropertyAccessor.php [hunk 3]
```diff
@@ -114,9 +109,33 @@ private function propertyExists($objectOrArray, $propertyPath)
...
+            if ($reflectionClass->hasProperty($propertyPath))
```
Comment (theofidry) [inline src/PropertyAccess/ReflectionPropertyAccessor.php]: same here (CS)

File: src/PropertyAccess/ReflectionPropertyAccessor.php [hunk 4]
```diff
@@ -48,15 +49,9 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
...
-            $setPropertyClosure = Closure::bind(
-                function ($object) use ($propertyPath, $value) {
-                    $object->{$propertyPath} = $value;
-                },
-                $objectOrArray,
-                $objectOrArray
-            );
-
-            $setPropertyClosure($objectOrArray);
+            $reflectionProperty = new ReflectionProperty($this->findClass($objectOrArray, $propertyPath), $propertyPath);
```
Comment (theofidry) [inline src/PropertyAccess/ReflectionPropertyAccessor.php]: your changes seems to account for `setValue()` but not `getValue()`

**Comment by jeremy-smith-maco** [comment]
@theofidry Hopefully it's good now. Couldn't get the unit testing setup on my local machine so hopefully it passes CI.

**Comment by theofidry** [comment]
Closing as per the discussion in #762.

**Comment by piotrbrzezina** [comment]
any progress ?

**Pull request closed without merging**